models/meshery_controllers.go

```diff
@@ -74,105 +74,117 @@ func NewMesheryControllersHelper(log logger.Handler, operatorDepConfig controlle
 // the meshsync data for that context is properly being handled
 func (mch *MesheryControllersHelper) UpdateMeshsynDataHandlers() *MesheryControllersHelper {
 	// only checking those contexts whose MesheryConrollers are active
-	mch.mu.Lock()
-	defer mch.mu.Unlock()
-	for ctxID, controllerHandlers := range mch.ctxControllerHandlersMap {
-		if _, ok := mch.ctxMeshsyncDataHandlerMap[ctxID]; !ok {
-			// brokerStatus := controllerHandlers[MesheryBroker].GetStatus()
-			// do something if broker is being deployed , maybe try again after sometime
-			brokerEndpoint, err := controllerHandlers[MesheryBroker].GetPublicEndpoint()
-			if brokerEndpoint == "" {
+	go func(mch *MesheryControllersHelper) {
+		mch.mu.Lock()
+		defer mch.mu.Unlock()
+		for ctxID, controllerHandlers := range mch.ctxControllerHandlersMap {
+			if _, ok := mch.ctxMeshsyncDataHandlerMap[ctxID]; !ok {
+				// brokerStatus := controllerHandlers[MesheryBroker].GetStatus()
+				// do something if broker is being deployed , maybe try again after sometime
+				brokerEndpoint, err := controllerHandlers[MesheryBroker].GetPublicEndpoint()
+				if brokerEndpoint == "" {
+					if err != nil {
+						mch.log.Warn(err)
+					}
+					mch.log.Info(fmt.Sprintf("skipping meshsync data handler setup for contextId: %v as its public endpoint could not be found", ctxID))
+					continue
+				}
+				mch.log.Info(fmt.Sprintf("found meshery-broker endpoint: %s for contextId: %s", brokerEndpoint, ctxID))
+				brokerHandler, err := nats.New(nats.Options{
+					// URLS: []string{"localhost:4222"},
+					URLS:           []string{brokerEndpoint},
+					ConnectionName: MesheryServerBrokerConnection,
+					Username:       "",
```

Comment:
Does Username and Password need to be explicitly set here? Whether an empty string is assumed or not, it would be good to externalize these variables (even if they are set to nil / empty).